tests/fixtures/mailbox_invalid.txt

```diff
@@ -111,13 +94,9 @@ aaa.com
 aaa@.com
 aaa@.123
 aaa@[123.123.123.123]a
-aaa@[123.123.123.333]
 a@bar.com.
-a@-b.com
-a@b-.com
 -@..com
 -@a..com
-invalid@about.museum-
```

Comment:
The `validate` function checks the validity of domains with the MX lookup and MX connection attempts. I'd argue that that's the correct place/way to check since the RFCs about address grammar don't say anything about the domain except that it should be a dot-atom or domain literal. For example `foo@bar` and `foo@com` are both perfectly valid addresses that will fail the MX lookup.